src/pages/scientificServices/pipelines/hooks/usePipelinesList.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react';
+import { Teaspoons } from 'src/libs/ajax/teaspoons/Teaspoons';
+import { Pipeline } from 'src/libs/ajax/teaspoons/teaspoons-models';
+import { notify } from 'src/libs/notifications';
+import { useCancellation } from 'src/libs/react-utils';
+
+export interface UsePipelinesListResult {
+  pipelines: Pipeline[];
+  isLoading: boolean;
+  error: Error | undefined;
+}
+
+export const usePipelinesList = (): UsePipelinesListResult => {
```

Comment:
factored this fetch logic out into a hook because we now have multiple components relying on it